From c30e410a1e262d92938bf372a3da09d18b8857f9 Mon Sep 17 00:00:00 2001
From: Philip Mueller <philm@manjaro.org>
Date: Fri, 31 Jul 2026 21:53:23 +0200
Subject: [PATCH 24/91] [FROM-ML] iio: imu: bmi270: Match PNP ID found on
 OneXPlayer X1 Pro (AMD)

The OneXPlayer X1 Pro (AMD) uses 10EC5280 rather than BMI0260 or BOSC0260 as PNP ID.
Adding the ID will enable the BMI260 accelerometer on the device using the bmi270 driver.

Closes: https://github.com/ublue-os/bazzite/issues/4805
Cc: stable@vger.kernel.org # 6.18+
Signed-off-by: Philip Mueller <philm@manjaro.org>
---
 drivers/iio/imu/bmi270/bmi270_i2c.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/iio/imu/bmi270/bmi270_i2c.c b/drivers/iio/imu/bmi270/bmi270_i2c.c
index d1168c4b03df..f5d10f27c9a5 100644
--- a/drivers/iio/imu/bmi270/bmi270_i2c.c
+++ b/drivers/iio/imu/bmi270/bmi270_i2c.c
@@ -46,6 +46,8 @@ static const struct acpi_device_id bmi270_acpi_match[] = {
 	{ "BMI0260",  (kernel_ulong_t)&bmi260_chip_info },
 	/* OrangePi Neo 2026 (since BIOS BCPH01_V1.20), etc. */
 	{ "BOSC0260", (kernel_ulong_t)&bmi260_chip_info },
+	/* OneXPlayer X1 Pro (AMD), etc. */
+	{ "10EC5280", (kernel_ulong_t)&bmi260_chip_info },
 	{ }
 };
 MODULE_DEVICE_TABLE(acpi, bmi270_acpi_match);
-- 
2.55.0

